src/Dfc.CourseDirectory.WebV2/Features/Venues/DeleteVenue/DeleteVenue.cs

```diff
@@ -0,0 +1,232 @@
+    }
+
+    public class ViewModel : Command
+    {
+        public string VenueName { get; set; }
+        public IReadOnlyCollection<string> AddressParts { get; set; }
+        public string PostCode { get; set; }
+    }
+
+    public class AffectedCourseViewModel
+    {
+        public Guid CourseId { get; set; }
+        public Guid CourseRunId { get; set; }
+        public string CourseName { get; set; }
+    }
+
+    public class AffectedApprenticeshipViewModel
+    {
+        public Guid ApprenticeshipId { get; set; }
+        public string ApprenticeshipName { get; set; }
+    }
+
+    public class AffectedTLevelViewModel
+    {
+        public Guid TLevelId { get; set; }
+        public string TLevelName { get; set; }
+    }
+
+    public class DeletedQuery : IRequest<DeletedViewModel>
+    {
+    }
+
+    public class DeletedViewModel
+    {
+        public string VenueName { get; set; }
+    }
+
+    public class Handler :
+        IRequestHandler<Query, OneOf<NotFound, ViewModel>>,
+        IRequestHandler<Command, OneOf<NotFound, ModelWithErrors<ViewModel>, Success>>,
+        IRequestHandler<DeletedQuery, DeletedViewModel>
+    {
+        private readonly ICosmosDbQueryDispatcher _cosmosDbQueryDispatcher;
+        private readonly ISqlQueryDispatcher _sqlQueryDispatcher;
+        private readonly IProviderOwnershipCache _providerOwnershipCache;
+        private readonly JourneyInstanceProvider _journeyInstanceProvider;
+
+        public Handler(
+            ICosmosDbQueryDispatcher cosmosDbQueryDispatcher,
+            ISqlQueryDispatcher sqlQueryDispatcher,
+            IProviderOwnershipCache providerOwnershipCache,
+            JourneyInstanceProvider journeyInstanceProvider)
+        {
+            _cosmosDbQueryDispatcher = cosmosDbQueryDispatcher;
+            _sqlQueryDispatcher = sqlQueryDispatcher;
+            _providerOwnershipCache = providerOwnershipCache;
+            _journeyInstanceProvider = journeyInstanceProvider;
+        }
+
+        public async Task<OneOf<NotFound, ViewModel>> Handle(Query request, CancellationToken cancellationToken)
+        {
+            var venue = await _cosmosDbQueryDispatcher.ExecuteQuery(new GetVenueById { VenueId = request.VenueId });
+
+            if (venue == null)
+            {
+                return new NotFound();
+            }
+
+            var getCourses = _cosmosDbQueryDispatcher.ExecuteQuery(new GetAllCoursesForProvider { ProviderUkprn = venue.Ukprn, CourseStatuses = CourseStatus.Live | CourseStatus.Pending | CourseStatus.BulkUploadPending | CourseStatus.BulkUploadReadyToGoLive | CourseStatus.APIPending | CourseStatus.APIReadyToGoLive | CourseStatus.MigrationPending | CourseStatus.MigrationReadyToGoLive });
+            var getApprenticeships = _cosmosDbQueryDispatcher.ExecuteQuery(new GetApprenticeships { Predicate = a => a.ProviderUKPRN == venue.Ukprn && a.ApprenticeshipLocations.Any(al => al.VenueId == request.VenueId) });
```

Comment:
are we going to have efficiency or RU issues with this?

Reply:
See above comment